### PR TITLE
fix Issue 22749 - importC: C11 does not allow taking the address of a bit-field

### PR DESCRIPTION
**Pull request by ibuclaw**
@WalterBright said that this wasn't possible at the last DLFM, I tried it, and DMD compiled it.

### CODE REVIEW

File: src/dmd/expressionsem.d [hunk 1]
```diff
@@ -6922,6 +6922,18 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
...
         }
+        if (auto dve = exp.e1.isDotVarExp())
+        {
+            /* https://issues.dlang.org/show_bug.cgi?id=22749
+             * Error about taking address of any bit-field, regardless of
+             * whether SCOPE.Cfile is set.
+             */
+            if (auto bf = dve.var.isBitFieldDeclaration())
+            {
+                exp.error("cannot take address of bit-field `%s`", bf.toChars());
```
Comment (maxhaton) [inline src/dmd/expressionsem.d:6933]: Supplemental error too?

Comment (maxhaton) [reply src/dmd/expressionsem.d:6933]: i.e. print where the bitfield is defined since you could be consuming C code from who knows where

**Comment by maxhaton** [comment]
Good spot

File: src/dmd/expressionsem.d (hunk 1 above)
Comment (ibuclaw) [reply src/dmd/expressionsem.d:6933]: Why? You're taking the address of a bit-field, enough said.

Comment (maxhaton) [reply src/dmd/expressionsem.d:6933]: I guess, just that context as to where the bitfield came from doesn't hurt

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**